pepper-apis/src/main/java/org/broadinstitute/ddp/event/dsmtask/impl/updateprofile/UpdateProfileProcessor.java

```diff
@@ -0,0 +1,41 @@
+package org.broadinstitute.ddp.event.dsmtask.impl.updateprofile;
+
+import org.broadinstitute.ddp.event.dsmtask.api.DsmTaskData;
+import org.broadinstitute.ddp.event.dsmtask.api.DsmTaskProcessorAbstract;
+
+/**
+ * Processes Dsm Task of taskType='UPDATE_PROFILE'.
+ * It updates person's email, first, and last name.
+ *
+ * <p>Does the following steps:
+ * <ul>
+ *     <li>updates user's email (checking if this email not used by another user);</li>
+ *     <li>updates user's first and last name.</li>
+ * </ul>
+ * NOTE: if any data (passed from Dsm) is null then it not updated.
+ */
+public class UpdateProfileProcessor extends DsmTaskProcessorAbstract {
+
+    public static final String TASK_TYPE__UPDATE_PROFILE = "UPDATE_PROFILE";
+
+    @Override
+    public void doIt(DsmTaskData dsmTaskData) {
+        UpdateProfileData updateProfileData = (UpdateProfileData)dsmTaskData.getPayloadObject();
+
+        updateEmail(dsmTaskData.getParticipantGuid(), updateProfileData.getEmail());
```

Comment:
Should we be wrapping both update calls in a try/catch that determines whether it should `ack()` or `nack()?`  For example, if there is a transient db or auth0 connectivity issue, it would be nice to `nack()` and use an exponential backoff (configured in the pubsub topic) to retry for a while.  Perhaps `doIt()` at the interface level should return some kind of enum with values like `SUCCESS`, `RETRY`, or `FAILURE` so that the dispatcher can apply the appropriate `ack()` or `nack()`?

Reply:
I see that we have a try/catch in the "abstract" processor class, which will return a "error" result data object. However, similar to Zim's concern, I don't see anywhere handling the error case and calling `nack()`. The "result sender" seems to go straight to publishing the result...

Reply:
Refactored this code: added exception DsmTaskException which have boolean flag meaning that if this exception thrown with flag set to true, then nack() should be called (i.e. it will try to process the task again).